src/Encryption.java

```diff
@@ -0,0 +1,49 @@
+package src;
+
+import java.security.SecureRandom;
+import java.security.spec.KeySpec;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+public class Encryption {
+
+    private static SecretKeyFactory factory;
+
+    public static SecretKey getKeyFromPassword(String password, String salt)
+            throws Exception {
+
+        factory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA256");
+        KeySpec spec = new PBEKeySpec(password.toCharArray(), salt.getBytes(), 65536, 256);
+        SecretKey secret = new SecretKeySpec(factory.generateSecret(spec)
+                .getEncoded(), "AES");
+        return secret;
+    }
+
+    public static SecretKey generateKey(int n) throws Exception {
+        KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
+        keyGenerator.init(n);
+        SecretKey key = keyGenerator.generateKey();
+        return key;
+    }
+
+    public static IvParameterSpec generateIv() {
+        byte[] iv = new byte[16];
+        new SecureRandom().nextBytes(iv);
+        return new IvParameterSpec(iv);
+    }
+
+    public static String encrypt(String algorithm, String input, SecretKey key,
+            IvParameterSpec iv) {
+        return "";
+    }
+
+    public static String decrypt(String algorithm, String cipherText, SecretKey key,
```

Comment:
remove static and make method abstract for AES/DES/RSA to provide the implementation